src/sync.js

```diff
@@ -508,10 +511,30 @@ SyncApi.prototype._sync = function(syncOptions) {
     // normal timeout= plus buffer time
     const clientSideTimeoutMs = this.opts.pollTimeout + BUFFER_PERIOD_MS;
```

Comment:
Nitpick: this should be the other side of the syncAccumulator stuff, before the `if (!isCachedResponse) {` where its used.